README.md

```diff
@@ -26,4 +26,66 @@ To test in a local Jenkins instance
 
     mvn hpi:run
 
-  [wiki]: http://wiki.jenkins-ci.org/display/JENKINS/SCM+API+Plugin
\ No newline at end of file
+  [wiki]: http://wiki.jenkins-ci.org/display/JENKINS/SCM+API+Plugin
+
+# SCM API Data Model
+
+## SCMSource
+
+[SCMSource](src/main/java/jenkins/scm/api/SCMSource.java)
+[(doc)](http://javadoc.jenkins.io/plugin/scm-api/jenkins/scm/api/SCMSource.html)
+implementations provide a configuration form (as a
+[Descriptor](http://javadoc.jenkins-ci.org/hudson/model/Descriptor.html))
+for users, and define the methods for multi-branch and other advanced
+SCM concepts in Jenkins.
+
+The most crucial methods an
+[SCMSource](src/main/java/jenkins/scm/api/SCMSource.java) must
+implement are:
+
+* [retrieve(HeadObserver, TaskListener)](https://github.com/jenkinsci/scm-api-plugin/blob/c06684f6e252bd4f8ef99ab8ec6eefceed8dce45/src/main/java/jenkins/scm/api/SCMSource.java#L153) - provides HeadObserver with the Set(name, hash) of revisions to consider
+* [retrieve(SCMHead, TaskListener)](https://github.com/MarkEWaite/scm-api-plugin/blob/c06684f6e252bd4f8ef99ab8ec6eefceed8dce45/src/main/java/jenkins/scm/api/SCMSource.java#L222) - returns a single revision
+* [retrieve(String, TaskListener)](https://github.com/MarkEWaite/scm-api-plugin/blob/c06684f6e252bd4f8ef99ab8ec6eefceed8dce45/src/main/java/jenkins/scm/api/SCMSource.java#L282) - returns a single revision identified by the string value
+
+## SCMHead
+
+[SCMHead](src/main/java/jenkins/scm/api/SCMHead.java)
+[(doc)](http://javadoc.jenkins.io/plugin/scm-api/jenkins/scm/api/SCMHead.html)
+is a named tree (subversion branch, mercurial branch, git branch,
+perforce branch/stream, etc.), identified by name
+
+## SCMRevision
+
+[SCMRevision](src/main/java/jenkins/scm/api/SCMRevision.java)
+[(doc)](http://javadoc.jenkins.io/plugin/scm-api/jenkins/scm/api/SCMRevision.html)
+is a specific change in the tree (subversion revision, mercurial hash,
+git sha, perforce change, etc.) identified by a version control
+specific object.
+
+## SCMStatus
+
+SCMStatus implementations (like
```

Comment:
I provide a new recommended path for this in the implementation guide. I recommend new SCM implementations do not follow this path, rather consolidate events through the SCM API's event system